src/terminal/scroll.lua

```diff
@@ -0,0 +1,150 @@
+--- Scroll Module.
+-- Provides functions to manage scroll regions and perform scrolling operations in the terminal.
+-- @module scroll
+-- @usage
+
+local output = require "terminal.output"
+
+local M = {}
+
+-- Internal scroll reset sequence
+local _scroll_reset = "\27[r"
+
+-- Stack to keep track of scroll regions
+local _scrollstack = {
+  _scroll_reset,
+}
+
+--=============================================================================
+-- Scroll Region Management (Non-Stack)
+--=============================================================================
+
+--- Creates an ANSI sequence to set the scroll region without writing.
+-- @tparam[opt] number top Top row of the scroll region.
+-- @tparam[opt] number bottom Bottom row of the scroll region.
+-- @treturn string ANSI sequence to set the scroll region.
+function M.scroll_regions(top, bottom)
+  if not top and not bottom then
+    return _scroll_reset
+  end
+  return "\27[" .. tostring(top) .. ";" .. tostring(bottom) .. "r"
+end
+
+--- Sets the scroll region and writes it to the terminal.
+-- @tparam[opt] number top Top row of the scroll region.
+-- @tparam[opt] number bottom Bottom row of the scroll region.
+-- @return true
+function M.scroll_region(top, bottom)
+  output.write(M.scroll_regions(top, bottom))
+  return true
+end
+
+--=============================================================================
+-- Basic Scrolling (Without Stack)
+--=============================================================================
+
+--- Creates an ANSI sequence to scroll up without writing to the terminal.
+-- @tparam[opt=1] number n Number of lines to scroll up.
+-- @treturn string ANSI sequence to scroll up.
+function M.scroll_ups(n)
+  n = n or 1
+  return "\27[" .. tostring(n) .. "S"
+end
+
+--- Scrolls the screen up and writes it to the terminal.
+-- @tparam[opt=1] number n Number of lines to scroll up.
+-- @return true
+function M.scroll_up(n)
+  output.write(M.scroll_ups(n))
+  return true
+end
+
+--- Creates an ANSI sequence to scroll down without writing to the terminal.
+-- @tparam[opt=1] number n Number of lines to scroll down.
+-- @treturn string ANSI sequence to scroll down.
+function M.scroll_downs(n)
+  n = n or 1
+  return "\27[" .. tostring(n) .. "T"
+end
+
+--- Scrolls the screen down and writes it to the terminal.
+-- @tparam[opt=1] number n Number of lines to scroll down.
+-- @return true
+function M.scroll_down(n)
+  output.write(M.scroll_downs(n))
+  return true
+end
+
+--- Creates an ANSI sequence to scroll vertically without writing to the terminal.
+-- @tparam[opt=0] number n Number of lines to scroll (negative for up, positive for down).
+-- @treturn string ANSI sequence for vertical scroll.
+function M.scroll_s(n)
```

Comment:
also an extraneous `_`
```suggestion
function M.scrolls(n)
```